tests/test-localAddress.js

```diff
@@ -1,27 +1,49 @@
 'use strict'
-
 var request = require('../index')
-  , tape = require('tape')
+    , tape = require('tape')
+
+tape('bind to invalid address', function (t) {
+    request.get({
+        uri: 'http://www.google.com',
+        localAddress: '1.2.3.4'
+    }, function (err, res) {
+        t.notEqual(err, null)
+        t.equal(err.message, 'bind EADDRNOTAVAIL')
+        t.equal(res, undefined)
+        t.end()
+    })
+})
 
-tape('bind to invalid address', function(t) {
-  request.get({
-    uri: 'http://www.google.com',
-    localAddress: '1.2.3.4'
-  }, function(err, res) {
-    t.notEqual(err, null)
-    t.equal(err.message, 'bind EADDRNOTAVAIL')
-    t.equal(res, undefined)
-    t.end()
-  })
+tape('bind to local address', function (t) {
+    request.get({
+        uri: 'http://www.google.com',
+        localAddress: '127.0.0.1'
+    }, function (err, res) {
+        t.notEqual(err, null)
+        t.equal(res, undefined)
+        t.end()
+    })
 })
 
-tape('bind to local address', function(t) {
-  request.get({
-    uri: 'http://www.google.com',
-    localAddress: '127.0.0.1'
-  }, function(err, res) {
-    t.notEqual(err, null)
-    t.equal(res, undefined)
-    t.end()
-  })
+tape('bind to local address on redirect', function (t) {
+    var os = require('os')
+    var localInterfaces = os.networkInterfaces()
+    var localIPS=[]
+    Object.keys(localInterfaces).forEach(function (ifname) {
+        localInterfaces[ifname].forEach(function (iface) {
+            if ('IPv4' !== iface.family || iface.internal !== false) {
+                // skip over internal (i.e. 127.0.0.1) and non-ipv4 addresses
+                return;
+            }
+                localIPS.push(iface.address);
+        });
+    });
+    request.get({
+        uri: 'http://google.com', //redirects to 'http://google.com'
```

Comment:
@0x4139 did you ever checked the travis-ci errors https://travis-ci.org/request/request/jobs/51369942 ?